src/unix.rs

```diff
@@ -81,61 +88,66 @@ impl Client {
         Ok(Client::from_fds(pipes[0], pipes[1]))
     }
 
-    pub unsafe fn open(s: &str) -> Option<Client> {
-        Client::from_fifo(s).or_else(|| Client::from_pipe(s))
+    pub unsafe fn open(s: &str) -> Result<Client, ErrFromEnv> {
+        match (Self::from_fifo(s), Self::from_pipe(s)) {
+            (Some(Ok(c)), _) | (_, Some(Ok(c))) => Ok(c),
+            (Some(Err(e)), _) | (_, Some(Err(e))) => Err(e),
+            (None, None) => Err(ErrFromEnv::ParseEnvVar),
+        }
     }
 
     /// `--jobserver-auth=fifo:PATH`
-    fn from_fifo(s: &str) -> Option<Client> {
+    fn from_fifo(s: &str) -> Option<Result<Client, ErrFromEnv>> {
```

Comment:
Would it be more intuitive with `Result<Option<Client>, ErrFromEnv>`? So that we can return `Err(…)` instead of `Some(Err(…))`.

Or even more, just `Result<Client, ErrFromEnv>` and return `Err(ParseEnvVar)` for the old `None` code path.

Reply:
Current `None` isn't actual error. Wouldn't it be less intuitive to use `Err` in this case? I'm not sure.

As for `Result<Option<Client>, ErrFromEnv>` - it's better, than my variant, if `Option` will remain here - I'll change to this.

Reply:
I proposed removing `Option` because [there](https://github.com/alexcrichton/jobserver-rs/pull/52/files#diff-52b718f0ee2e4e8580f7155fe871a4267a6427607bf486005bed17325f066a1dR95) with `(None, None)` we actually convert into `ParseEnvVar`. I feel like they are a bit inconsistent. Why is it an error here but not here?

Reply:
Sounds reasonable! I am sold 😆

Just realized there is no button "Mark as Resolved" for this thread even it was initiated by me 🤔